### PR TITLE
Ziti hosting

### PR DESCRIPTION
**Pull request by ekoby**
adds bind/listen/accept support on Ziti sockets
add `@openziti.zitify()` decorator to simplify hosting configuration
lazy start and load of ziti indetities

### CODE REVIEW

File: src/openziti/decor.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+def _patchedSocket(patch_opts):
+    class patchedSocket(zitisock.ZitiSocket):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **dict(kwargs, opts=patch_opts))
+    return patchedSocket
+
+
+class MonkeyPatch():
+    def __init__(self, **args):
```
Comment (sabedevops) [inline src/openziti/decor.py:32]: Is this intentionally `**args` ("**dict") as opposed to `*args` ("*tuple")?

File: src/openziti/decor.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for m_name, _ in self.orig_methods.items():
+            sock.__dict__[m_name] = self.orig_methods[m_name]
+
+
+
+
+def zitify(**zargs):
```
Comment (sabedevops) [inline src/openziti/decor.py:50]: Here also. In python, "args" typically refers to the tuple of positional arguments to a function. Should this be `**zkwargs`?

File: src/openziti/zitisock.py [hunk 3]
```diff
@@ -40,14 +44,40 @@ def connect(self, addr) -> None:
...
+                zitilib.connect_addr(self._zitifd, addr)
+            except:
                 PySocket.close(self)
                 self._zitifd = None
                 PySocket.__init__(self, self._ziti_af, self._ziti_type,
                                   self._ziti_proto)
                 PySocket.connect(self, addr)
 
+    def bind(self, addr) -> None:
+        from .context import get_context
```
[the remark this answers is not in the document]
Comment (ekoby) [reply src/openziti/zitisock.py:57]: no, this overrides socket.bind() method

[the remark this answers is not in the document]
Comment (sabedevops) [reply src/openziti/zitisock.py:57]: At import time?

File: src/openziti/decor.py (hunk 2 above)
Comment (ekoby) [reply src/openziti/decor.py]: renamed

**Pull request merged**